mypy/checkexpr.py

```diff
@@ -1452,6 +1452,33 @@ def check_call_expr_with_callee_type(
         object_type: Type | None,
         member: str | None = None,
     ) -> Type:
+
+        proper_callee = get_proper_type(callee_type)
+
+        is_constructor_call = False
+        if isinstance(e.callee, RefExpr):
+            node = e.callee.node
+            if node is not None and hasattr(node, "name"):
+                is_constructor_call = node.name == "__init__"
+            elif callable_name is None:
+                # direct class call without member name
+                is_constructor_call = True
+
+        object_type = get_proper_type(object_type)
+        if (
+            isinstance(proper_callee, CallableType)
+            and isinstance(object_type, Instance)
+            and is_constructor_call
+        ):
+            target_name = object_type.type.name
+            arg_names_set = set(e.arg_names or [])
+
+            for name, kind in zip(proper_callee.arg_names, proper_callee.arg_kinds):
+                if name is not None and kind in (ARG_NAMED, ARG_POS):
+                    if name not in arg_names_set:
+                        if target_name == "misc":
+                            continue  # Skip error for miscellaneous/unknown classes
```

Comment:
Many questions here... First, why is this code block preceding a docstring? 

Second, there's no special-cased "misc" typing construct, this simply shouldn't exist. But fortunately this whole block starting from `is_constructor_call = ...` line is... doing nothing? Is this some kind of merge artifact?  I don't see any use of these newly declared variables.

Reply:
Since you requested my review, this comment still applies - I don't see any reason for this whole block of code to exist...